### DISCUSSION TITLE
keyword `remove` for dynamic object to expose TryDeleteMember

### DISCUSSION THREAD

**Discussion by Thaina** [dotnet/csharplang#143] [General]
For manipulating dynamic object for interop with dynamic language or other service by json or any other dynamic data format. Ability to delete member is necessary to completely satisfied contract, making data that exclude the member instead of null

And it already supported by `TryDeleteMember` API of `DynamicObject`. Only need keyword to expose that functionality

So I would suggest reusing `remove` keyword in contextually in front of dynamic object

```C#
dynamic obj = new JsonObject();
obj.X = 0;
delete obj.X;
```

dynamic object cannot be assumed that it could use function member. So we can't use `dynamicObj.Remove` or anything like that. Because it will be ambiguous if that object also have member in that name. So keyword to operate on it is necessary

### Alternative

We could introduce new keyword. Normally it would be `delete`. I just think we already have `remove` being keyword so it would be less likely to be used than `delete`

**Comment by miloush** [comment]
> I just think we already have `remove` being keyword so it would be less likely to be used than `delete` 

That actually makes it much likely to be chosen over introducing a new one.

Besides that, I would find `delete` very confusing/controversial because of memory allocation connotations.

**Comment by Thaina** [comment]
@miloush I mean we might have each of those word being used as variable name in the past

```C#
var delete = Http.Request("DELETE",url).Response();
if(delete.HttpStatuscode == 200)
    //dosomething;
```

But `remove` is being keyword already so it less likely to be used like that

**Comment by miloush** [comment]
I understand your concern, but that is usually not a problem, since there is no situation in which you could use the variable the way you propose the keyword to be used. This has been done many times in the past, for example with the `dynamic` keyword.

On the other side, I do see the value in matching the name of the method.
